sr_finger_mount/src/sr_finger_mount.py

```diff
@@ -0,0 +1,157 @@
+    CONST_FREQ_MIN = 5
+    CONST_FREQ_MAX = 30
+
+    CONST_PST_MAX = 200  # 1600 for real PSTs
+    CONST_PST_MIN = 1  # 350 for real PSTs
+
+    CONST_FINGERS = ["ff", "mf", "rf", "lf", "th"]
+    CONST_ACCEPTABLE_DEVICE_NAMES = ["Boreas DevKit", "BOS1901-KIT"]
+
+    def __init__(self, fingers, hand_id="rh"):
+        self._used_fingers = fingers
+        self._used_devices = []
+
+        self._sub = rospy.Subscriber("/"+hand_id+"/tactile", ShadowPST, self._tactile_cb)
+
+        self._am = dict()
+        self._fm = dict()
+
+        if not set(self._used_fingers).intersection(set(self.CONST_FINGERS)):
+            rospy.logwarn("Verify used fingers!")
+            return
+
+        if self.check_devices():
+            self.init_all()
+
+    def _tactile_cb(self, data):
+        if len(data.pressure) == 5:
+            mapped_pst_values = dict(zip(self.CONST_FINGERS, data.pressure))
+
+            for f in self._used_fingers:
+                self._am[f] = ((mapped_pst_values[f]-self.CONST_PST_MIN)/(self.CONST_PST_MAX-self.CONST_PST_MIN)) * \
+                              (self.CONST_AMP_MAX-self.CONST_AMP_MIN)+self.CONST_AMP_MIN
+                self._fm[f] = ((mapped_pst_values[f]-self.CONST_PST_MIN)/(self.CONST_PST_MAX-self.CONST_PST_MIN)) * \
+                              (self.CONST_FREQ_MAX-self.CONST_FREQ_MIN)+self.CONST_FREQ_MIN
+        else:
+            rospy.logwarn("Missing values")
+
+    def check_devices(self):
+        needed_devices = math.ceil(len(self._used_fingers)//2)
+        device_list = sd.query_devices()
+        present_devices = 0
+
+        for acceptable_device in self.CONST_ACCEPTABLE_DEVICE_NAMES:
+            for device in device_list:
+                if acceptable_device in device['name']:
+                    self._used_devices.append(device['name'])
+                    present_devices += 1
+
+        if needed_devices > present_devices:
+            rospy.logerr("Not enough dev kits ({}/{}) connected to handle {} fingers".format(present_devices,
+                                                                                             needed_devices,
+                                                                                             len(self._used_fingers)))
+            return False
+        return True
+
+    def _sublists(self):
+        channels_per_device = 2
+        for i in range(0, len(self._used_fingers), channels_per_device):
+            yield self._used_fingers[i:i + channels_per_device]
+
+    def init_all(self):
+        device_handlers = [None, None]
+        finger_sets = list(self._sublists())
+        for i, finger_set in enumerate(finger_sets):
+            device_name = self._used_devices[i]
+            device_handlers[i] = DeviceHandler(device_name, finger_set, self)
+            time.sleep(1)
+            device_handlers[i].start()
+
+
+if __name__ == "__main__":
+
+    fingers = rospy.get_param("~fingers", 'th')
+    side = rospy.get_param("~side", 'rh')
```

Comment:
our convention is typically side: right, left, hand_id: rh, lh. So please use "hand_id" here

Reply:
Corrected